compiler/rustc_arena/src/lib.rs

```diff
@@ -103,7 +104,7 @@ impl<T> ArenaChunk<T> {
                 // A pointer as large as possible for zero-sized elements.
                 ptr::invalid_mut(!0)
             } else {
-                self.start().add((*self.storage.as_ptr()).len())
+                self.start().add(self.storage.as_ptr().len())
```

Comment:
These are nice changes, but the lint is not able to suggest them, right? The lint would have suggested `&*`.

Reply:
Yes. We could probably see if the parent expression is a method call and the pointer also has a method with the same name...

Reply:
Seems like a good option to keep in mind for the future. :) But maybe shouldn't be in the MVP.

Reply:
Sure, I'll make a follow-up then :)